### ISSUE TITLE
View Components cannot be included, mystery exception references irrelevant model.

### ISSUE THREAD

**Issue by janis-veinbergs** [aspnet/Razor#2342]
This is a follow up to https://github.com/aspnet/Docs/issues/6278 A quest trying to figure out what is the default search path for ViewComponents

Probably I am doing something wrong, but I cannot get ViewComponent to be included in a page. 

I have a folder structure:
```
- Pages
  - Components
    - ExternalSystems
      - Default.cshtml
- Views
  - Shared
    - Components
      - ExternalSystems
        - Default.cshtml
```

within `Pages/Index.cshtml` i'v tried to include this component by trial & error (i'm learning this stuff after all):
```
@await Component.InvokeAsync("ExternalSystems");
@await Component.InvokeAsync("/Pages/Components/ExternalSystems");
@await Component.InvokeAsync("/Pages/Components/ExternalSystems/Default");
@await Component.InvokeAsync("/Pages/Components/ExternalSystems/Default.cshtml");
@await Component.InvokeAsync("/Views/Shared/ExternalSystems");
```

I expected that exception will show places searched like it does for `@Html.Partial`:
> InvalidOperationException: The partial view 'none' was not found. The following locations were searched: /Pages/none.cshtml /Views/Shared/none.cshtml

However for every single call to `@await Component.InvokeAsync` it spits this exception: `InvalidOperationException: Could not find an 'Invoke' or 'InvokeAsync' method for the view component '<Stripped.Out>.Pages.Identities.IndexModel'.`
First of all, it doesn't show me paths searched. Second, wait what? Why are you telling me something about Pages.Identities. Yes, I have that model, but it is nowhere referenced in neither `Pages/Index.cshtml` or View Component i'm trying to include.

Could someone please provide me with a guidance to determine: How view component paths are searched? For Razor Pages it is not documented, only in [3rd party site](https://www.learnrazorpages.com/razor-pages/view-components). 
Anyway it doesn't work that way for me - what would be the debugging steps? Console debug doesn't show anything useful.

dotnet 2.0.7

**Comment by mkArtakMSFT** [comment]
Hi. It looks like this is a question about how to use ASP.NET Core. While we do our best to look through all the issues filed here, to get a faster response we suggest posting your questions to [StackOverflow](https://www.stackoverflow.com) using the **asp.net-core-mvc** tag.

**Comment by janis-veinbergs** [comment]
@mkArtakMSFT you are right. I should have thought of that. Thank you. Given that answer will be provided there, I will post it here too.
https://stackoverflow.com/questions/50266952/how-are-razor-page-view-component-paths-searched-i-fail-to-get-them-working-n

**Comment by mkArtakMSFT** [comment]
Thanks @janis-veinbergs!

**Comment by janis-veinbergs** [comment]
Alright, my mistake was that I didn't see I have decorated `...Pages.Identities.IndexModel` with `[ViewComponent]`. It was unintentional and was just trying to make ViewComponents work by trial & error and failed to see this mistake.

Basically dotnet discovered type that didn't have either Invoke/InvokeAsync function and thus was stuck on that exception.

I validated that these paths are searched when providing "HelloWorld" as view name:

- /Pages/Components/HelloWorld/HelloWorld.cshtml
- /Views/Shared/Components/HelloWorld/HelloWorld.cshtml
